### PR TITLE
Add explicit logging for assaytype lookup failures + Fix endpoint

### PR DESCRIPTION
**Pull request by NickAkhmetov**
This PR adds error logging to display the parameters of requests to the assaytype service.

### CODE REVIEW

File: context/app/api/client.py [hunk 1]
```diff
@@ -211,7 +211,17 @@ def get_vitessce_conf_cells_and_lifted_uuid(self, entity, marker=None, wrap_erro
                 def get_assaytype(entity):
                     uuid = entity.get('uuid')
                     url = f"{current_app.config['SOFT_ASSAY_ENDPOINT']}/{uuid}"
-                    return requests.get(url, headers=self._get_headers()).json()
+                    headers = self._get_headers()
+                    try:
+                        request = requests.get(url, headers=headers)
```
Comment (lchoy) [inline context/app/api/client.py:219]: This is a nitpick, but technically, requests.get() returns a requests.Response object.
```suggestion 
                        response = requests.get(url, headers=headers)
```

File: context/app/api/client.py [hunk 2]
```diff
@@ -211,7 +211,17 @@ def get_vitessce_conf_cells_and_lifted_uuid(self, entity, marker=None, wrap_erro
...
+                        return request.json()
+                    except Exception as e:
```
Comment (lchoy) [inline context/app/api/client.py:221]: The exception is raised when the response doesn't contain json. Should there be a check for the response status?

**Review by lchoy — CHANGES_REQUESTED** [review]

File: context/app/api/client.py (hunk 1 above)
Comment (NickAkhmetov) [reply context/app/api/client.py]: Done - correct/logical naming is important for clarity 😄

File: context/app/api/client.py [hunk 3]
```diff
@@ -211,7 +211,17 @@ def get_vitessce_conf_cells_and_lifted_uuid(self, entity, marker=None, wrap_erro
...
+                        current_app.logger.error(
+                            f'Failed to fetch from {url} with headers {headers}: {e}'
```
[the remark this answers is not in the document]
Comment (NickAkhmetov) [reply context/app/api/client.py]: Done; I've added logic to update `Authorization`'s value to `REDACTED` if it's present in the headers.

File: context/app/api/client.py (hunk 2 above)
Comment (NickAkhmetov) [reply context/app/api/client.py:221]: I figured we could implicitly handle any errors thrown by `requests.get` by including it in the try/except and catching all `Exception`s, not just the `JSONDecodeError` we were encountering before: https://docs.python-requests.org/en/latest/_modules/requests/exceptions

I have added `status` to the logged error, and also updated the error to be logged in the form of an object to make it easier to identify fields.

**Review by lchoy — APPROVED** [review]

**Pull request merged**